File: Rust/src/histogram.rs
// https://www.codewars.com/kata/59f44c7bd4b36946fd000052/rust

pub fn hist(s: &str) -> String
{
	let mut strings: Vec<String> = vec![];
	for letter in ['u', 'w', 'x', 'z']
	{
		let count = s.matches(letter).count();
		if count < 1
		{
			continue;
		}
		let string = format!("{0}  {1:<5} {2}", letter, count, "*".repeat(count));
		strings.push(string);
	}
	return strings.join("\n");
}

File: JavaScript/src/histogram.js
// https://www.codewars.com/kata/59f44c7bd4b36946fd000052/javascript

export class Histogram
{
	/** @param {string} s
	 * @returns string */
	static hist(s)
	{
		const count = {};
		for (const letter of s)
		{
			count[letter] = count[letter] ? count[letter] + 1 : 1;
		}
		
		let res = [];
		for (const letter of "uwxz")
		{
			if (count[letter])
			{
				const num = `${count[letter]}      `.substring(0, 6);
				const stars = "*".repeat(count[letter]);
				res.push(`${letter}  ${num}${stars}`);
			}
		}
		
		return res.join("\n");
	}
}
